android/src/main/java/com/oakam/launchdarkly_flutter/LaunchdarklyFlutterPlugin.java

```diff
@@ -85,6 +86,33 @@ public void onDetachedFromActivity() {
 
   }
 
+  private LDUser createUser(@NonNull MethodCall call) {
+    LDUser.Builder userBuilder;
+
+    if (call.hasArgument("userKey")) {
+      String userKey = call.argument("userKey");
+      userBuilder = new LDUser.Builder(userKey);
+    } else {
+      userBuilder = new LDUser.Builder(UUID.randomUUID().toString()).anonymous(true);
```

Comment:
If you switch user context, it will fetch the values for the new user. Also, at least for Android:

> the SDK will store the last 5 user contexts on a single device, with support for switching between different user contexts.

Regarding UUID for anonymous user – yeah, it's actually a preferred way:

> You must generate a unique key for anonymous users. Session IDs or UUIDs work best for this.